src/internal/fl-patch.js

```diff
@@ -0,0 +1,22 @@
+var fl = require('./fl.js');
+var fixedAp = function(f) {
+  return f.ap(this);
+};
+
+var patch = function(obj){
+  return Object.keys(fl).forEach(function(key) {
+    if (typeof obj[key] === 'function') {
+      if (key === 'ap') {
+        obj[fl[key]] = fixedAp;
```

Comment:
This doesn't feel right to me. What would be the consequences of removing this exception?